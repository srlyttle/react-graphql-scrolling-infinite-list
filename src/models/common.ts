export interface Repository {
  name: string;
  description: string;
  owner: { login: string };
  stargazers: { totalCount: number };
  primaryLanguage: { name: string };
}

export interface RepositoriesProps {
  loading: boolean;
  repos: Repository[];
  loadMoreRepos: Function;
}

export interface SearchData {
  search?: SearchDataContainer;
}

export interface SearchDataContainer {
  edges: Edge[];
  pageInfo: PageInfo;
  repositoryCount: number;
  __typename: string;
}

export interface PageInfo {
  endCursor: string;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
  startCursor: string;
  __typename: string;
}

export interface Edge {
  node: RepositoryDetails;
  __typename: string;
}

export interface RepositoryDetails {
  description: string;
  name: string;
  owner: RepositoryOwner;
  primaryLanguage: PrimaryLanguage;
  stargazers: StargazerItem;
  __typename: string;
}
export interface RepositoryOwner {
  login: string;
  __typename: string;
}

export interface PrimaryLanguage {
  name: string;
  __typename: string;
}
export interface StargazerItem {
  totalCount: number;
  __typename: string;
}
