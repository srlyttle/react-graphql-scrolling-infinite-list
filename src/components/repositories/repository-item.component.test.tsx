import React from 'react';

import { render } from '@testing-library/react';
import Repositories from './repository-item.component';
import { RepositoryDetails } from '../../models/common';

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

const repo: RepositoryDetails = {
  name: 'genetic-drawing',
  owner: { login: 'anopara', __typename: 'User' },
  description: 'A genetic algorithm toy project for drawing',
  stargazers: { totalCount: 1286, __typename: 'StargazerConnection' },
  primaryLanguage: { name: 'Python', __typename: 'Language' },
  __typename: 'Repository',
};
test('<Repository Item is rendered correctly', async () => {
  const { container } = render(<Repositories repositoryDetails={repo} />);
  expect(container).toMatchSnapshot();
});
