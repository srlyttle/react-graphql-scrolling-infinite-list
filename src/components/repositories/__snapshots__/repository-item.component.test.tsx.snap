// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository Item is rendered correctly 1`] = `
<div>
  <li>
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            genetic-drawing - anopara
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card ant-card-bordered ant-card-type-inner"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                A genetic algorithm toy project for drawing
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            Python
          </div>
        </div>
        <div
          class="ant-card ant-card-bordered ant-card-type-inner"
          style="margin-top: 16px;"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Stars
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            1286
          </div>
        </div>
      </div>
    </div>
  </li>
</div>
`;
